src/main/scala/encry/view/wallet/AccountManager.scala

```diff
@@ -0,0 +1,63 @@
+package encry.view.wallet
+
+import encry.crypto.encryption.AES
+import encry.crypto.{PrivateKey25519, PublicKey25519}
+import encry.settings.{Algos, WalletSettings}
+import io.iohk.iodb.{ByteArrayWrapper, Store}
+import scorex.crypto.encode.Base58
+import scorex.crypto.hash.Blake2b256
+import scorex.crypto.signatures.{Curve25519, PrivateKey, PublicKey}
+import scorex.utils.Random
+
+case class AccountManager(store: Store, settings: WalletSettings) {
+
+  import encry.storage.EncryStorage._
+
+  def mandatoryAccount: PrivateKey25519 = store.get(AccountManager.MandatoryAccountKey).flatMap { res =>
+    store.get(AccountManager.AccountPrefix +: res.data).map { secretRes =>
+      PrivateKey25519(PrivateKey @@ AES.decrypt(secretRes.data, settings.password), PublicKey @@ res.data)
+    }
+  } getOrElse createMandatory(settings.seed.flatMap(seed => Base58.decode(seed).toOption))
+
+  def accounts: Seq[PrivateKey25519] = store.getAll().foldLeft(Seq.empty[PrivateKey25519]) { case (acc, (k, v)) =>
```

Comment:
Maybe it's better to persist keys for all accounts and retrieve accounts by it instead of retrieving everything from iodb and filtering it?